### PR TITLE
Initial support for Windows 10 client detection (including Windows 10 Enterprise edition)

### PR DESCRIPTION
**Pull request by cycocrew**

### CODE REVIEW

File: jcl/source/common/JclSysInfo.pas [hunk 1]
```diff
@@ -259,7 +259,7 @@ function GetShellProcessHandle: THandle;
     weWinVistaHomePremium, weWinVistaBusiness, weWinVistaBusinessN,
     weWinVistaEnterprise, weWinVistaUltimate, weWin7Starter, weWin7HomeBasic,
     weWin7HomePremium, weWin7Professional, weWin7Enterprise, weWin7Ultimate,
-    weWin8, weWin8Pro, weWin8Enterprise, weWin8RT, weWin81, weWin81Pro, weWin81Enterprise, weWin81RT);
+    weWin8, weWin8Pro, weWin8Enterprise, weWin8RT, weWin81, weWin81Pro, weWin81Enterprise, weWin81RT, weWin10, weWin10Enterprise);
```
Comment (obones) [inline jcl/source/common/JclSysInfo.pas]: I believe Win10 should start on a newline

File: jcl/source/common/JclSysInfo.pas [hunk 2]
```diff
@@ -251,7 +251,7 @@ function GetShellProcessHandle: THandle;
    (wvUnknown, wvWin95, wvWin95OSR2, wvWin98, wvWin98SE, wvWinME,
     wvWinNT31, wvWinNT35, wvWinNT351, wvWinNT4, wvWin2000, wvWinXP,
     wvWin2003, wvWinXP64, wvWin2003R2, wvWinVista, wvWinServer2008,
-    wvWin7, wvWinServer2008R2, wvWin8, wvWin8RT, wvWinServer2012, wvWin81, wvWin81RT, wvWinServer2012R2);
+    wvWin7, wvWinServer2008R2, wvWin8, wvWin8RT, wvWinServer2012, wvWin81, wvWin81RT, wvWinServer2012R2, wvWin10);
```
Comment (obones) [inline jcl/source/common/JclSysInfo.pas]: I believe Win10 should start on a new line

File: jcl/source/common/JclSysInfo.pas [hunk 3]
```diff
@@ -3368,8 +3369,13 @@ function GetWindowsVersion: TWindowsVersion;
                 else
                   Result := wvWinServer2012R2;
               end;
+          end;
         end;
-        end;
+        10:
```
Comment (obones) [inline jcl/source/common/JclSysInfo.pas]: Win10 really is major version 10? I mean win8 has major version equal to 6...

**Comment by cycocrew** [comment]
Additional changes made to detect Windows 10 client correctly. As Windows 8.1, Windows 10 also requires a workaround as Win32MajorVersion shows up as 6 (instead of 10) and Win32MinorVersion shows up as 2 (instead of 0). Windows 10 is supposed to be 10.0, Windows 8.1 is supposed to be 6.3 and Windows 8 is 6.2.

File: jcl/source/common/JclSysInfo.pas (hunk 3 above)
Comment (cycocrew) [inline jcl/source/common/JclSysInfo.pas]: Windows 10 kernel version is 10.0 in Windows 10 Technical Preview Build 9926. It was 6.4 in previous builds.

**Comment by cycocrew** [comment]
Additional changes made to detect Windows 10 client correctly. It appears that GetVersion(Ex) is deprecated starting with Windows 8.1, explaining why it sends back Windows 8 (6.2) version. Alternative detection methods have been implemented.

**Comment by obones** [comment]
Hum, I believe it's not actually deprecated but that rather it only works if the manifest is telling that the application supports 8.1: https://msdn.microsoft.com/en-us/library/windows/desktop/dn302074%28v=vs.85%29.aspx
Having workaround methods is nice, though

**Comment by cycocrew** [comment]
Exact wording from https://msdn.microsoft.com/en-us/library/windows/desktop/dn302074.aspx:

"Manifestation

In Windows 8.1, the GetVersion(Ex) APIs have been deprecated. That means that while you can still call the APIs, if your app does not specifically target Windows 8.1, you will get Windows 8 versioning (6.2.0.0).

Solution

In order to target Windows 8.1, you need to either include the app manifest or include _NT_TARGET_VERSION=$ (_NT_TARGET_VERSION_LATEST) in the source file."

**Comment by cycocrew** [comment]
Is there anything left preventing this pull request to be merged?

**Comment by melloware** [comment]
Agreed where are we with merging this change.  windows 10 went live last month.

**Pull request closed without merging**